lib/python/rose/job_runner.py

```diff
@@ -183,36 +183,53 @@ def run(self, job_manager, *args):
             self.job_processor.post_process_job(job_proxy, *args)
 
         """
-        nproc = self.nproc
-        if nproc > len(job_manager.jobs):
-            nproc = len(job_manager.jobs)
-        pool = Pool(processes=nproc)
-
-        results = {}
-        while job_manager.has_jobs():
-            # Process results, if any is ready
-            for name, result in results.items():
-                if result.ready():
-                    results.pop(name)
-                    job_proxy, args_of_events = result.get()
-                    for args_of_event in args_of_events:
-                        self.job_processor.handle_event(*args_of_event)
-                    job = job_manager.put_job(job_proxy)
-                    if job_proxy.exc is None:
-                        self.job_processor.post_process_job(job, *args)
-                        self.job_processor.handle_event(JobEvent(job))
-                    else:
-                        self.job_processor.handle_event(job_proxy.exc)
-            # Add some more jobs into the worker pool, as they are ready
-            while job_manager.has_ready_jobs():
-                job = job_manager.get_job()
-                if job is None:
-                    break
-                job_run_args = [self.job_processor, job] + list(args)
-                result = pool.apply_async(_job_run, job_run_args)
-                results[job.name] = result
-            if results:
-                sleep(self.POLL_DELAY)
+#         nproc = self.nproc
```

Comment:
I had real problems getting multiprocessing.pool to work: @mattshin had to disable it, at least for now, and this section will need to be re-written in  a python3 friendly way.

The problem seems to be caused by python 3 being unable to serialize some objects and therefore to pass them to processes. :(

Reply:
Hey! Just wanted to lyk that you probably meant to tag someone else in this comment 😅 - maybe @mshin?

Reply:
Yes!

Reply:
Thank you @mattshin, I did mean @mshin